pkg/engine/cmd.go

```diff
@@ -64,6 +65,12 @@ func (e *Engine) runCommand(ctx context.Context, tool types.Tool, input string)
 	cmd.Stderr = io.MultiWriter(all, os.Stderr)
 	cmd.Stdout = io.MultiWriter(all, output)
 
+	if isCredential {
+		pause := context2.GetPauseFuncFromCtx(ctx)
+		unpause := pause()
+		defer unpause()
```

Comment:
This could technically be simplified down to `defer context2.GetPauseFuncFromCtx(ctx)()()` but I was afraid that would be too confusing, so I split it into three lines to make it clearer what is happening here.

Reply:
Part of me would love to see this as `defer context2.GetPauseFuncFromCtx(ctx)()()`, but I think the split is the right idea.

Reply:
And then you'd be a terrible person. What maniac would do a chained function call in a single line defer :)